File: Frontend/src/modules/admin/feature/customer/services/customer.service.ts
import api from "@/common/api";

import type {
  CustomerResponse,
  CustomerSearchParams,
  PageResponse,
} from "../types/customer.types";

const buildParams = (params: CustomerSearchParams) => {
  const result: Record<string, string | number> = {};

  if (params.keyword && params.keyword.trim() !== "") {
    result.keyword = params.keyword.trim();
  }

  if (params.status !== null && params.status !== undefined) {
    result.status = params.status;
  }

  result.page = params.page ?? 0;
  result.size = params.size ?? 10;

  return result;
};

export const customerService = {
  getCustomers(params: CustomerSearchParams) {
    return api.get<PageResponse<CustomerResponse>>("/admin/customers", {
      params: buildParams(params),
    });
  },

  getCustomerById(userId: number) {
    return api.get<CustomerResponse>(`/admin/customers/${userId}`);
  },

  updateStatus(userId: number, status: number) {
    return api.patch<CustomerResponse>(
      `/admin/customers/${userId}/status`,
      null,
      {
        params: { status },
      }
    );
  },
};

File: Backend/src/main/java/org/example/datn_sd69/modules/customer/service/CustomerService.java
package org.example.datn_sd69.modules.customer.service;

import org.example.datn_sd69.modules.customer.dto.response.CustomerResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CustomerService {

    Page<CustomerResponse> getCustomers(String keyword, Integer status, Pageable pageable);

    CustomerResponse getCustomerById(Integer userId);

    CustomerResponse updateStatus(Integer userId, Integer status);
}
